lession/classes-and-objects

AHTI6IOTIK opened the pull request.

Commit:
Выполнил задание по классам и объектам

iStemmer commented on src/classes/and/objects/AuthorNotFilled.java:
```diff
@@ -0,0 +1,4 @@
+package classes.and.objects;
+
+public class AuthorNotFilled extends Exception {
```
Мы это еще не приходили. Как не проходили ООП =). 
Но наследоваться от Exception  это слишком, так как это слишком общее исключение (в таких случаях обычно наследуются от RuntimeException и не указывают их в методах, потому что Runtime unchecked) и тем более слишком широкое - чтобы его указывать в методе конструктора (Такое вообще не нужно делать). Если ты указываешь исключение - тебе его нужно как-то обрабатывать, а иначе зачем ты его выбрасываешь? То есть нужен код, который поймает AuthorNotFilled и обработает правильно вывод. 
Exception удалить.

AHTI6IOTIK replied:
По п6 я должен отклонить создание объекта без автора. 
Перевел на RuntimeException добавил try/catch

iStemmer commented on src/classes/and/objects/Book.java:
```diff
@@ -0,0 +1,46 @@
+    Author author;
+    int publishYear;
+
+    public Book(String name, Author author, int publishYear) throws AuthorNotFilled {
```
Удалить

AHTI6IOTIK replied:
Убрал эксепшен из сигнатуры

iStemmer commented on src/classes/and/objects/Book.java:
```diff
@@ -0,0 +1,46 @@
+import java.util.Objects;
+
+public class Book {
+    String name;
```
не хватает модификатора private для полей класса. Предполагается что мы будем работать с геттерами и сеттерами, а поля задавать в конструкторе. Если оставить поля класса открытыми - необходимость в сеттерах отпадает, более того - любой код сможет поменять поля минуя сеттеры.

AHTI6IOTIK replied:
Добавил модификатор доступа

iStemmer commented on src/classes/and/objects/Author.java:
```diff
@@ -0,0 +1,34 @@
+package classes.and.objects;
+
+public class Author {
+    String firstName;
```
модификатор private (см комментарий ниже)

AHTI6IOTIK replied:
добавил модификатор доступа

Commit:
внес правки по ревью

- убрал эксепшени из сигнатуры
- добавил try/catch
- эксепшн отнаследовал от RuntimeException
